hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,762 @@
+],
+  ["ByteOrder=1",
+  {name: "sd[0] (host output)", wave: "x22222222222|2222|222|22x", data: ["t[7]", "t[6]", "t[5]", "t[4]", "t[3]", "t[2]", "t[1]", "t[0]", "t[15]","t[14]",
+                                                                          "t[13]","t[9]","t[8]","t[23]","t[22]","t[16]", "t[31]", "t[30]", "t[25]", "t[24]"]},
+  {name: "sd[1] (host input)", wave: "x22222222222|2222|222|22x", data: ["r[7]", "r[6]", "r[5]", "r[4]", "r[3]", "r[2]", "r[1]", "r[0]", "r[15]","r[14]",
+                                                                         "r[13]","r[9]","r[8]","r[23]","r[22]","r[16]", "r[31]", "r[30]", "r[25]", "r[24]"]},
+  {name: "Which Byte?", wave: "x5.......5..|..5.|.5.|..x", data: ["LSB from TXDATA/RXDATA", "","", "          MSB"]}
+],
+  ],
+  head: {
+    text: "Serial bit ordering for 32-bit TXDATA & RXDATA data words (t[31:0] and r[31:0]), Standard SPI as a Function of the Parameter 'ByteOrder'",
+  },
+  foot: {
+  text: "(Bits are numbered as they appear when loaded into TXDATA or RXDATA registers.)"
+  }
+}
+{{< /wavejson >}}
+
+
+As shown in the following figure, a similar time-ordering scheme applies for Dual- and Quad-mode transfers.
+However many bits of similar significance are packed into multiple parallel `sd` data lines, with the most significant going to `sd[0]`.
+
+{{< wavejson >}}
+{signal: [
+  ["ByteOrder=0",
+  {name: "sd[0]", wave: "x...22334455x...", data: ["d[31]", "d[27]", "d[23]", "d[19]", "d[15]", "d[11]", "d[7]", "d[3]"]},
+  {name: "sd[1]", wave: "x...22334455x...", data: ["d[30]", "d[26]", "d[22]", "d[18]", "d[14]", "d[10]", "d[6]", "d[2]"]},
+  {name: "sd[2]", wave: "x...22334455x...", data: ["d[29]", "d[25]", "d[21]", "d[17]", "d[13]", "d[9]", "d[5]", "d[1]"]},
+  {name: "sd[3]", wave: "x...22334455x...", data: ["d[28]", "d[24]", "d[20]", "d[16]", "d[12]", "d[8]", "d[4]", "d[0]"]},
+],
+   ["ByteOrder=1",
+  {name: "sd[0]", wave: "x...55443322x...", data: ["d[7]", "d[3]", "d[15]", "d[11]", "d[23]", "d[19]", "d[31]", "d[27]"]},
+  {name: "sd[1]", wave: "x...55443322x...", data: ["d[6]", "d[2]", "d[14]", "d[10]", "d[22]", "d[18]", "d[30]", "d[26]"]},
+  {name: "sd[2]", wave: "x...55443322x...", data: ["d[5]", "d[1]", "d[13]", "d[9]", "d[21]", "d[17]", "d[29]", "d[25]"]},
+  {name: "sd[3]", wave: "x...55443322x...", data: ["d[4]", "d[0]", "d[12]", "d[8]", "d[20]", "d[16]", "d[28]", "d[24]"]},
+  ],
+  ],
+  head: {
+   text: "Serial bit ordering for 32-bit data word (d[31:0]), Quad SPI as a Function of the Parameter 'ByteOrder'",
+  },
+  foot: {
+  text: "(Bits are numbered as they appear when loaded into TXDATA or RXDATA registers.)"
+  }
+}
+{{< /wavejson >}}
+
+### Command Length and Alignment in TXDATA,RXDATA
+
+There is no restriction on the length of a command.
+Even though the TXDATA and RXDATA registers require 4-byte accesses, any unused bytes in the TXDATA fifo (or unused slots in the RXDATA fifo), are simply ignored at the end of a command.
```

Comment:
@tjaychen: this whole section from #5077 has been moved here to help close it.

[Your comment on this line](https://github.com/lowRISC/opentitan/pull/5077#discussion_r571315714) has been carried over here also:

> hmmm.... this could end up becoming a problem.
> We can talk to the software guys a bit more..but i'm pretty sure they really don't like the idea of having to pack in software to 4-byte alignments. I'm pretty certain the ideal would be to treat this write almost like a memory.
> 
> I guess the point you're trying to make is that they don't need to pack, the bytes un-written will simply not be used?
> This could end up wasting a bit of space.. but i guess it's not a huge problem. Let me know if i understood correctly.